File: app/src/main/java/com/essence/mvvmapplication/di/post/PostComponent.java
package com.essence.mvvmapplication.di.post;

import com.essence.mvvmapplication.posts.ui.view.PostsActivity;

import dagger.Subcomponent;

@Subcomponent
@PostScope
public interface PostComponent {
    @Subcomponent.Builder
    interface Builder {
        PostComponent create();
    }

    void inject(PostsActivity postsActivity);
}

File: app/src/main/java/com/essence/mvvmapplication/di/post/PostComponent.kt
package com.essence.mvvmapplication.di.post

import com.essence.mvvmapplication.posts.ui.view.PostsActivity
import dagger.Subcomponent

@Subcomponent
@PostScope
interface PostComponent {
    @Subcomponent.Builder
    interface Builder {
        fun create(): PostComponent?
    }

    fun inject(postsActivity: PostsActivity?)
}